lib/validator/v3/validator.go

```diff
@@ -954,6 +955,23 @@ func validateNetworkPolicy(structLevel validator.StructLevel) {
 	}
 }
 
+func validateNetworkSet(structLevel validator.StructLevel) {
```

Comment:
Is this going to work if someone does a get, modify, update type set of operations?

Won't the GET return an object with a namespace label on it? Maybe we just need to filter out that label on GET.

Reply:
Or alternatively only return an error if it doesn't match the Namespace field? WDYT?

Reply:
As discussed, we'll leave this check in and simplify the insertion of the `projectcalico.org/namespace` label.